### PR TITLE
Silvio/dpc 666 add git hook

### PR DESCRIPTION
**Pull request by SMLuthi**
### Fixes [DPC-666](https://jiraent.cms.gov/browse/DPC-666)

Avoid accidentally pushing a commit with sensitive vars declared in the env files for postman. 

### Proposed Changes

Create a pre-commit hook that checks the env and globals json env vars and validates that the values are set to an empty string.

Include a makefile that allows a developer to quickly install the git hook(s)

### Change Details

A pre-commit script within a new git hooks folder

Makefile to install the git-hook(s)

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### Acceptance Validation

If the pre-commit hook script encounters a stored env var that isnt supposed to be commited:
Screen Shot 2020-09-16 at 4 41 15 PM

If the pre-commit hook script encounters invalid json in a `.json` file thats being checked:
Screen Shot 2020-09-17 at 4 36 30 PM

If the pre-commit hook script encounters a missing file thats being checked:
Screen Shot 2020-09-17 at 4 44 42 PM

### CODE REVIEW

**Review by Sun-Mountain — APPROVED** [review]
Works for me!

File: .git-hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Only need to check the files if they have been committed with changes
+FILES_MODIFIED = `git diff --cached --name-only`.chomp.split("\n")
+
+errors = []
+
+FILES_MODIFIED.each do |file|
+  if ENV_FILES.key?(file)
+    file_data = JSON.parse(File.read(file))
+    file_data['values'].each do |v|
+      if ENV_FILES[file].include?(v['key']) && !v['value'].empty?
```
Comment (dhgreene) [inline .git-hooks/pre-commit:45]: Can we add a little more logic here?  Arguable, if the JSON content of these files is not parsable, the error message is useful to the dev:

```
 > git commit -m "invalid JSON"

=======================================================
Running Verification for tokens, secret keys, etc...
=======================================================

Traceback (most recent call last):
	4: from .git-hooks/pre-commit:41:in `<main>'
	3: from .git-hooks/pre-commit:41:in `each'
	2: from .git-hooks/pre-commit:43:in `block in <main>'
	1: from /Users/davidgreene/.rvm/rubies/ruby-2.6.3/lib/ruby/2.6.0/json/common.rb:156:in `parse'
/Users/davidgreene/.rvm/rubies/ruby-2.6.3/lib/ruby/2.6.0/json/common.rb:156:in `parse': 767: unexpected token at 'organization_id: asdf (JSON::ParserError)
'
```

But if the file is missing entirely, I'd prefer not to have my commit blocked:

```
> git commit -m "missing file"
=======================================================
Running Verification for tokens, secret keys, etc...
=======================================================

Traceback (most recent call last):
	3: from .git-hooks/pre-commit:41:in `<main>'
	2: from .git-hooks/pre-commit:41:in `each'
	1: from .git-hooks/pre-commit:43:in `block in <main>'
.git-hooks/pre-commit:43:in `read': No such file or directory @ rb_sysopen - postman/DPC_API.postman_globals.json (Errno::ENOENT)
```

Comment (SMLuthi) [reply .git-hooks/pre-commit]: @dhgreene These change requests have been added in! Check the screenshots above to see the results.

**Review by Sun-Mountain — APPROVED** [review]

**Review by dhgreene — APPROVED** [review]
🤖

**Pull request merged**